src/main/scala/cromwell/engine/backend/io/filesystem/gcs/GcsFileSystem.scala

```diff
@@ -29,16 +34,18 @@ object GcsFileSystem {
   */
 class GcsFileSystem private (gcsFileSystemProvider: GcsFileSystemProvider, gcsRoot: String) extends FileSystem {
 
+
+
   import GcsFileSystem._
 
-  val root = gcsRoot match {
-    case GsUriRegex(chunks) => new NioGcsPath(chunks.split(Separator), true)(this)
-    case _ => throw new InvalidPathException(gcsRoot, s"Root of GCS file system must be an absolute GCS path: $gcsRoot")
+  val root: Option[Path] = gcsRoot match {
```

Comment:
Are there Specs you can point me to that explain _why_ we have to keep track of different filesystem objects with different roots? This variable is everywhere, but I can't figure out what functionality it actually provides.

Ideally I'd like to get rid of the `root` as a variable all together. If there's some way of having a singleton `GcsFileSystem` I'd love to see it.